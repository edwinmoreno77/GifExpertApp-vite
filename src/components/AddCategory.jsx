import { useState } from "react"
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }
    return (
        <form onSubmit={e => onSubmit(e)} aria-label="form">
            <input type="text"
                placeholder="Search Gif"
                value={inputValue}
                onChange={e => handleInputChange(e)}

            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
