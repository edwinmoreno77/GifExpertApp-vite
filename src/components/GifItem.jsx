import PropTypes from 'prop-types';


export const GifItem = ({ title, url, }) => {
    return (
        <div className="card animate__animated animate__heartBeat">
            <img src={url} alt={title} />
            <p className="p"><b>{title}</b></p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}