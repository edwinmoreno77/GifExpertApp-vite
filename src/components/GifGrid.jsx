import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';





export const GifGrid = ({ cat }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(cat);

        setImages(newImages);
    }

    useEffect(() => {
        getImages(cat);

    }, [])

    return (
        < >
            <h2>{cat}</h2>
            <div className='card-grid'>
                {
                    images.map((img) => (
                        <GifItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}