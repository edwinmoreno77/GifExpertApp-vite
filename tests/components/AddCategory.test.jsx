import { render, screen, fireEvent } from '@testing-library/react';
import { AddCategory } from '../../src/components/AddCategory';


describe('<AddCategory tests ', () => {

    test('should render AddCategory', () => {
        const { container } = render(<AddCategory onNewCategory={() => { }} />);
        // screen.debug();
        expect(container).toMatchSnapshot();

    }
    );

    test('should change the value of the text box', () => {
        render(<AddCategory onNewCategory={() => { }} />);
        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: 'saitama' } });
        expect(input.value).toBe('saitama');

    });

    test('should call onNewCategory if the input has a value', () => {

        const inputValue = 'saitama';
        render(<AddCategory onNewCategory={() => { }} />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue } });
        fireEvent.submit(form);
        expect(input.value).toBe('');

    });

});

