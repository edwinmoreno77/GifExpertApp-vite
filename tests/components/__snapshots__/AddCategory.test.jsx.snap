// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory tests  should render AddCategory 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
